packages/api-server/src/utils/db.ts

```diff
@@ -33,82 +32,21 @@ export function getConnectionOptions(options?: PoolOptions) {
     return Object.assign(defaultOptions, options);
 }
 
-export interface QueryResponse<T> {
-    result: T;
-    fields: FieldPacket[];
-}
-
-export class ConnectionWrapper {
-    public conn: Connection;
-    private log: pino.Logger;
-
-    constructor(options: PoolOptions) {
-        this.log = pino({ level: process.env.LOG_LEVEL || "info" });
-        this.conn = createConnection(options);
-        const errorHandler = (err: any) => {
-            if (!err.fatal) return;
-            if (err.code !== 'PROTOCOL_CONNECTION_LOST') throw err;
-
-            this.log.warn(`Database server connection lost, trying to reconnect.`);
-            const newConn = createConnection(this.conn.config);
-            newConn.connect();
-            newConn.on('error', errorHandler);
-            this.conn = newConn;
-        }
-        this.conn.on('error', errorHandler);
-    }
-
-    query<T extends RowDataPacket[][] | RowDataPacket[] | OkPacket | OkPacket[] | ResultSetHeader>(
-        sql: string,
-        values: any | any[] | { [param: string]: any }
-    ): Promise<QueryResponse<T>> {
-        return new Promise((resolve, reject) => {
-            this.conn.query<T>(sql, values, (err: QueryError | null, result, fields) => {
-                if (err) {
-                    reject(err);
-                } else {
-                    resolve({
-                        result, fields
-                    });
-                }
-            });
-        });
+export async function getConnection(options: ConnectionOptions, log: pino.Logger) {
```

Comment:
I think `getConnectionOptions` and `getConnection` both should be a method of fastify instance. And, `getConnectionOptions` should read config from `app.config` instead of from `process.env` directly.

Reply:
OK